### ISSUE TITLE
Using variable matchers to lint a more generic object

### ISSUE THREAD

**Issue by kylegillen** [schoero/eslint-plugin-readable-tailwind#62]
Hi. Great plugin! 
I went through the docs and saw the examples that target specific variables and object properties.

Wondering if / how I could get the plugin to lint this structure:

```jsx
  const buttonStyles = {
    base: 'ui-rounded-lg px-md ui-h-11 ui-px-sm ui-transition-all ui-text-base ui-shadow-sm ui-flex-1',
    disabled: 'disabled:ui-bg-brand-primary-disabled disabled:ui-cursor-not-allowed',
    primary: 'ui-bg-brand-primary ui-text-brand-onprimary hover:ui-bg-brand-primary-hover uif',
  }
```

Essentially, the variables will always end in `Styles` (e.g. `buttonStyles`), and I would like the matcher to apply to all properties within that object.

I tried the following, but as the type of matcher suggests, the `name` property is simply a string(?) not a regex, and my regex skills sadly aren't polished enough to match all the properties / values within the object.

```ts
variables: [[
  // matches variables endting in `Styles`
  '.*Styles',
  // matches the object value for the `base` key (ideally, should match ALL values for ALL keys).
  [
    {
      match: 'objectValues',
      pathPattern: '^base|\\.base',
    },
  ],
]],
```

I appreciate any help you could lend.
Thanks again!

**Comment by schoero** [comment]
Unfortunately, changing the names to regular expressions is a breaking change. 

I have a few other updates planned for v2. Once those are ready, I’ll include this change to release all breaking changes together.

**Comment by schoero** [comment]
Names can now be regular expressions in v2. Check the [updated docs](https://github.com/schoero/eslint-plugin-readable-tailwind/blob/main/docs/concepts/concepts.md#name) to learn more.
